patches/components-sync-engine_impl-syncer.h.patch

```diff
@@ -0,0 +1,21 @@
+diff --git a/components/sync/engine_impl/syncer.h b/components/sync/engine_impl/syncer.h
+index 23b389eada81c850c3a8856678d046e599a060da..48654f4c09558ea4d0ac246ebe16178019b30a0e 100644
+--- a/components/sync/engine_impl/syncer.h
++++ b/components/sync/engine_impl/syncer.h
+@@ -32,7 +32,7 @@ class SyncCycle;
+ // A Syncer instance expects to run on a dedicated thread.  Calls to SyncShare()
+ // may take an unbounded amount of time because it may block on network I/O, on
+ // lock contention, or on tasks posted to other threads.
+-class Syncer {
++class Syncer : public base::SupportsWeakPtr<Syncer> {
```

Comment:
better to add a WeakPtrFactory in `BRAVE_SYNCER_H_`

Reply:
that requires another `weak_ptr_factory_(this)` patch in the constructor, is that ok?

Reply:
crap, I guess that won't be any better in terms of adding vs modifying lines because they're on one line `: cancelation_signal_(cancelation_signal), is_syncing_(false) {}` so I guess this is fine

Reply:
but why do we need this to be WeakPtr again?

Reply:
for this
https://github.com/brave/brave-core/pull/2010/files#diff-aca217062a05c75ca9645dce7a37922fR26

Reply:
can we listen for `SyncServiceObserver::OnSyncShutdown` there instead?

Reply:
actually shouldn't it be be fine here? https://github.com/brave/brave-core/pull/2010/files#diff-586f16ca5da4f68977eb7a16ff267041R52
since sync_engine_impl is already a weak handle?

Reply:
addressed in b6f353557be0b4fc8d611c56c304025baf6356bd